### ISSUE TITLE
Questions/Suggestions on the data generated by AudioRecorder.startStream

### ISSUE THREAD

**Issue by jamescrosswell** [llfbandit/record#167]
Very cool package! 

I'm keen to leverage the new streaming support to listen for notes played by a musical instrument. I want to detect what note is played in near real time (so don't want or need a file). 

I grabbed the Beta and am just dipping my toes in the water... So far as I can work out, `AudioRecorder.startStream` creates a stream and then I guess I should setup a listener to actually get data from the stream for processing.

1. It might be nice to have both of those things happen in the same operation. This seems to be how [audio_streamer have implemented this](https://github.com/cph-cachet/flutter-plugins/blob/5606be32334777f9e7760eda796bfc7a961a9265/packages/audio_streamer/example/lib/main.dart#L47).
2. I'm not 100% sure how to interpret the Stream. Since it spits out 8 bit values but most encoders are 16 bit, I'm guessing the values come in pairs that need to be combined to reconstitute the original 16 bit sample?

In a muck around project, I currently have this wired up but I have no idea whether it's accurate or not. 

```
  Future<void> processStream(Stream<Uint8List> stream) async {
    // Assuming the PCM data is 16-bit little-endian
    const int bytesPerSample = 2;
    const bool isLittleEndian = true;

    stream.listen((Uint8List data) {
      // Process and print each chunk of Uint8List data
      for (int i = 0; i < data.lengthInBytes; i += bytesPerSample) {
        int sample = 0;
        for (int j = 0; j < bytesPerSample; j++) {
          int byte = isLittleEndian ? data[i + j] : data[i + bytesPerSample - j - 1];
          sample |= byte << (j * 8);
        }
        // Process the PCM sample (interpret as signed 16-bit integer)
        int pcmValue = sample.toSigned(16);
        // Do something with the PCM value
        log("$pcmValue");
      }      
    }, onDone: () {
      log('Stream ended');
    }, onError: (error) {
      log('Error: $error');
    });        
  }
```

Again, audio_streamer seems to do a little pre-processing with the stream so that package users can deal with `List<double>` (presumably amplitude values)... which is nice - you can then start processing the data immediately, rather than having to decode it. Potentially, if there was some reason folks might be interested in the raw 8 bit pairs, it could be done as a helper extension method on Uint8List.

**Comment by llfbandit** [comment]
Thanks for your contribution.

For your concern about missing stream events, this is not the case as the controller will buffer the events before attaching your listener. So you're safe on this. Still need to check if I'm not creating broadcast controller though. 

About streaming data type. I need some time to confirm this but you're right. Bounding data to `Uint8List` makes pcm usage, well... fuzzy and harder to handle.

Definitely will think about this improvement and clarification too for beta 2.

**Comment by jamescrosswell** [comment]
Thanks for the fast reply. 

I'm new to Dart and also to dealing with Audio, so I may be off on a tangent but I tracked down another package called `wav` that looked like it might help me to use the raw Uint8List data coming out of record. That package was really designed for reading/writing Wav files... but basically had some code for reading header information from those files and then some other code for reading the actual audio data... so I teased out the code for reading the data in a fork of the code that allows reading raw audio data. 

In [this example](https://github.com/jamescrosswell/wav/blob/raw-stream-reader/example/raw.dart), I've used that to read some raw audio from a file on disk (which kind of defeats the purpose... but it's just a POC). All that code really needs to be supplied, in order to work, is the stuff that you would ordinarily get from the header then:  
- bit depth
- number of channels
- samplesPerSecond

One question I had about your library (so that I can try to glue these things together):
- When using `AudioRecorder.startStream`, how can I determine those 3 values above? Is it something that can be supplied/configured before starting the stream? Is it ever dependent on the platform where the software is running (e.g. different for Android, iOS, Web etc.)?

**Comment by llfbandit** [comment]
Every encoders are aligned on 16 bits depth.
The number of channels is what your input is capable of + the configuration you assigned to `numChannels`. For your use case, you should stick to mono I guess. If you set up more channels, let's say 2, there's no conversion from mono to stereo from this plugin.
samplesPerSeconds should be equal to `sampleRate` which is expressed in Hertz.

For your convenience, an utility method will be added in beta 2: `convertBytesToInt16`.

**Comment by ivanesi** [comment]
Is it possible to get `List<double>` from the stream? I need to process data to fftea package that require List<double>
